### ISSUE TITLE
isAssignableFrom differences on 1.0.0-RC2

### ISSUE THREAD

**Issue by cquiroz** [scala-js/scala-js#3924]
While trying to make shapeless work with java 1.0.0-RC2 I found a case where the JVM and JS 0.6.x behaves the same but 1.0.0-RC2 don't.
for `Character` and `Long` this operations returns `true` in 0.6 and `false` in 1.0
```
    val c: Any = 'c'
    classOf[java.lang.Character].isAssignableFrom(c.getClass)}
```
The following repo contains a bit more code
https://github.com/cquiroz/assignable

/cc @milessabin

**Comment by sjrd** [comment]
Thanks for the report!

**Comment by sjrd** [comment]
So this can be minimized further to inlining `classOf[java.lang.Character]` instead of `c.getClass`. So this erroneously returns `false`:
```scala
classOf[java.lang.Character].isAssignableFrom(classOf[java.lang.Character])
```
The same is reproducible with `java.lang.Long` and `java.lang.Void` (which is `Unit`'s boxed class), but not for other boxed classes.

**Comment by sjrd** [comment]
Fixed in 46bb8439786382e71b1ab7b6611e6ba33ee9f783.
